app/models/unique_name.rb

```diff
@@ -5,17 +5,19 @@
 ###
 
 class UniqueName < ApplicationRecord
-
   def self.update!
     Rails.logger.info 'Updating the unique names table'
 
     # Build double metaphone representations for all names in the database
-    names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.map{ |n| n&.downcase || ''} + GrdaWarehouse::Hud::Client.source.select('LastName').distinct.pluck('LastName')&.map{ |n| n&.downcase || ''}
-    names.uniq.each do |name|
+    existing_names = UniqueName.pluck(:name)
+    all_names = GrdaWarehouse::Hud::Client.source.distinct.pluck('FirstName').map { |n| n.to_s.downcase } +
+      GrdaWarehouse::Hud::Client.source.distinct.pluck('LastName').map { |n| n.to_s.downcase }
```

Comment:
I would maybe write these as `GrdaWarehouse::Hud::Client.source.distinct.where.not(FirstName: [nil, '']).pluck('FirstName').map(&:downcase)`

Reply:
✅ Thanks for the tip. I've made and tested the adjustment.